src/App.js

```diff
@@ -0,0 +1,131 @@
+    });
+    const [arrOfTasks, setArrOfTasks] = useState([]);
+    const [completedTasks, setCompletedTasks] = useState([]);
+    const [inProgressTasks, setInProgressTasks] = useState([]);
+    const [isFilterMenu, setIsFilterMenu] = useState(false);
+    const [statusList, setStatusList] = useState({
+        activeCompleted: false,
+        activeInProgress: false,
+        activeAll: false,
+        completed: false,
+        inProgress: false,
+        all: false
+    });
+
+    useEffect(() => {
+        const tableData = JSON.parse(localStorage.getItem('tableData'));
+        setArrOfTasks(tableData);
+    }, []);
+
+    useEffect(() => {
+        localStorage.setItem('tableData', JSON.stringify(arrOfTasks));
+    }, [arrOfTasks]);
+
+    const taskObj = {
+        id: arrOfTasks.length + 1,
+        title: taskData.title,
+        description: taskData.description,
+        status: 'in progress'
+    };
+
+    const changeStatus = (id) => {
+        const newArr = arrOfTasks;
+        for (let i = 0; i < newArr.length; i++) {
+            if (newArr[i].id === id) {
+                newArr[i].status = 'completed';
+            }
+        }
+
+        return setArrOfTasks([...newArr]);
+    };
+
+    const addNewRowInTable = () => {
+        const newArr = [...arrOfTasks, taskObj];
+
+        (taskObj.title && taskObj.description) === '' ?
+          window.alert('The fields should not be empty')
+          : setArrOfTasks(newArr);
+
+        taskData.title = '';
+        taskData.description = '';
+    };
+
+    const removeTask = (id) => {
+        const newArr = arrOfTasks.filter((row) => row.id !== id);
+        setArrOfTasks(newArr);
+    };
+
+    const showCompletedTask = () => {
+        const newArr = arrOfTasks.filter((row) => row.status === 'completed');
+        setCompletedTasks(newArr);
+    };
+
+    const showInProgressTask = () => {
+        const newArr = arrOfTasks.filter((row) => row.status === 'in progress');
+        setInProgressTasks(newArr);
+    };
+
+    return (
+      <Context.Provider value={{ removeTask, changeStatus }}>
+          <div className="App">
+              <h1>Todo List</h1>
+              <div className="mainFilter">
+                  {arrOfTasks.length !== 0 ?
+                    <div onClick={() => setIsFilterMenu(!isFilterMenu)}>
+                        <FilterIcon/>
+                    </div> : null}
+                  <div className="filterList">
+                      {isFilterMenu ?
+                        <FilterList isCompletedStatus={showCompletedTask}
+                                    isInProgressStatus={showInProgressTask}
+                                    statusList={statusList}
+                                    setStatusList={setStatusList}
+                        /> : null}
+                  </div>
+              </div>
+              <TableList
+                data={(statusList.all || (statusList.completed && statusList.inProgress)) ? arrOfTasks :
```

Comment:
Move filtering logic to the separate function (create something like actualTodos which display just filtered list)